docker/Dockerfile

```diff
@@ -1,44 +1,85 @@
 ARG PYTHON_VER
-################################ Stage: base
-# Install all OS package upgrades and dependencies once from which we will base other containers
+ARG PYUWSGI_VER
+
+################################ Overview
+# This builds the following hierarchy of images:
+#
+# python:${PYTHON_VER}-slim --> (base) --> (install) --> (install-dev) -->  dev
+#                                 |            .              |
+#                                 V            .              V
+#                               final <.........          final-dev
+#
+# base (intermediate build target; basis for all other images herein)
+#   adds OS-level packaging dependencies for *running* Nautobot
+#   installs Pip and wheel
+#
+# install (intermediate build target)
+#   adds OS-level packaging dependencies for *installing* Nautobot, its dependencies, and its development dependencies
+#   installs Poetry
+#   uses Pip and Poetry to install Nautobot's Python dependencies (but *not* development dependencies)
+#   uses Poetry to build a Nautobot wheel
+#
+# install-dev (intermediate build target)
+#   uses Poetry to additionally install Nautobot's *development* dependencies
+#   adds files for Nautobot development server to run
+#
+# final-dev (development environment for Nautobot plugins)
+#   uses Pip to install Nautobot as a wheel
+#
+# dev (development environment for Nautobot core)
+#   uses Poetry to install Nautobot as editable
+#     (at runtime the requirement is to mount the live Nautobot source code to /source/; it won't run without that)
+#
+# final (production-ready environment):
+#   adds "nautobot" system user
+#   copies all installed Python packages, including Nautobot itself, from "install-nautobot"
+#   creates a self-signed SSL certificate
+#   sets up the system to run Nautobot in production with uwsgi
+#
+
+################################ Stage: base (intermediate build target; basis for all other images herein)
+
 FROM python:${PYTHON_VER}-slim as base
 
 ENV PYTHONUNBUFFERED=1 \
     NAUTOBOT_ROOT=/opt/nautobot \
     prometheus_multiproc_dir=/prom_cache
 
-# Install all OS package upgrades and dependencies
+# Install all OS package upgrades and dependencies needed to run Nautobot in production
 # hadolint ignore=DL3005,DL3008,DL3013
 RUN apt-get update && \
     apt-get upgrade -y && \
-    apt-get install --no-install-recommends -y git mime-support curl libxml2 libmariadb3 && \
+    apt-get install --no-install-recommends -y git mime-support curl libxml2 libmariadb3 openssl && \
```

Comment:
openssl was already being pulled into `base` as an indirect dependency; I just wanted to go ahead and make it explicit since we depend on it to generate the SSL certificates.